server/src/stateManagement.mjs

```diff
@@ -700,17 +700,39 @@ function setMethodError(userId, methodName, code, message) {
   });
 }
 
+/* if "group" keyword is passed for scope inside ctx functions, 
+and if userid invoking the YAML contains group, the group of the userId will be used as the "scope" for that call
+else,  the full userid is used along with warning
+*/
+function setScopeForGroupKeyword(userId, scope) {
+  let parseUserId = parseUser(userId)
+  if (parseUserId.group == undefined || parseUserId.group == null) {
```

Comment:
minor, but in JS, you can just test for falsiness... "if ( ! parseUserId.group ) {"
also minor, but probably "parsedUserId" (note the 'd')